tests/set_test.go

```diff
@@ -113,6 +113,11 @@ func TestSetWithOptions(t *testing.T) {
 			commands: []string{"SET k v XX EX 1", "GET k", "SLEEP 2", "GET k", "SET k v XX EX 1", "GET k"},
 			expected: []interface{}{"(nil)", "(nil)", "OK", "(nil)", "(nil)", "(nil)"},
 		},
+		{
```

Comment:
Add test cases for the following:
1. Validate that SET succeeds with GET option even if the key does not exist.
2. Validate error is returned when using SET with GET option if the previously stored value was not a string.
3. Validate previous value is returned when using SET with the GET option.

Reply:
Sure, I'll add these test cases 👍

Reply:
@JyotinderSingh , actually, I've some doubts:

1. like for the 1st one, in redis it's given that if the key doesn't exists, then it should return `(nil)`?
![image](https://github.com/user-attachments/assets/38dc88ed-803a-4f5b-9d55-633a4f1c5b34)

2. Also, could you please guide that how can I validate that if the previous values was a string or not? Because dice stores everything as string, am I right? Like `2` will be stored as `"2"`?
3. I've got the 3rd point. I'll add that 👍

Reply:
1. Even though it returns nil, the key is still set in the background. You return nil since there was no existing value, however, the new value will still be set.
2. Add the validation regardless.